### DISCUSSION TITLE
how to update a graph based on selected filter

### DISCUSSION THREAD

**Discussion by cgx-avenue** [zauberzeug/nicegui#2674] [Q&A] [answered]
### Description

Dear,

I'm new to NiceGUI and try to develop a basic demo. Now I encountered a basic need and searched but still don't know how to do it.
like shown in the pic, I use a selector to select a code as filter. After selection, the table will automatically show the corresponding transaction details. Below the table is a line chart that I want to use to show the transaction prices.
My target is simple: after I select the code in the selector, the table and graph will automatically update based on selected value.
For table, I used bind_filter_from and it works. But for the graph below, I haven't found any solution. Could you kindly help me on how to realize this? Any hint is welcome! Thanks!
![2024-03-07_11-07](https://github.com/zauberzeug/nicegui/assets/10353881/7e31aea8-0bb9-4efd-b550-c0a6ca54b5cd)
 Have a nice day!

**Comment by falkoschindler** [comment, marked as answer]
Hi @cgx-avenue,

You can simply handle the "value changed" event of the input element:

```py
from random import random
import plotly.graph_objects as go
from nicegui import ui

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'field': 'age'},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol', 'age': 42},
]

def update_plot(filter_value: str = '') -> None:
    fig.data = []
    for row in rows:
        if filter_value.lower() in row['name'].lower():
            fig.add_trace(go.Scatter(x=[1, 2, 3], y=[random(), random(), random()], name=row['name']))
    fig.update_layout(showlegend=True)
    plot.update()

name = ui.input('Name', placeholder='Your name', on_change=lambda e: update_plot(e.value))
ui.table(columns=columns, rows=rows, row_key='name').bind_filter_from(name, 'value')
fig = go.Figure()
plot = ui.plotly(fig)
update_plot()
```

**Reply by cgx-avenue** [reply]
Thanks for the quick and wonderful reply!
I modified the code based on your reply and now it works!
Thanks for the help!
